handler/src/main/java/io/netty/handler/ssl/SniHandler.java

```diff
@@ -306,6 +306,16 @@ public void operationComplete(Future<SslContext> future) throws Exception {
     }
 
     /**
+     * The default implementation will simply call {@link AsyncMapping#map(Object, Promise)} but
+     * users can override this method to implement custom behavior.
+     *
+     * @see AsyncMapping#map(Object, Promise)
+     */
+    protected Future<SslContext> lookup(ChannelHandlerContext ctx, String hostname) throws Exception {
```

Comment:
Either works, do you have a preference? I think `protected` is better or maybe a getter method for it.